src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -86,8 +63,35 @@
   If the query can't be parsed, a 400 is returned."
   [entity version query paging-options db url-prefix]
   (try
-    (let [parsed-query (json/parse-strict-string query true)]
-      (stream-query-result entity version parsed-query paging-options db url-prefix http/stream-json-response))
+    (let [parsed-query (json/parse-strict-string query true)
+          query-error (promise)
+          count-query (promise)
```

Comment:
I wanted to keep the api with `streamed-query-results` semi consistent so that we could make small changes to the extensions repo. The problem is essentially that we try to do too much in streamed-query-results, we do a query->sql, which then returns possibly two "queries", one which we then proceed to execute inside the streamed-query-results function, and the other which we really only need if we're doing a http query, the count-query. In a future world, I think we need a different scheme for all this, separating streamed-query-results from query->sql (which i have taken steps towards here by moving the entity-map stuff out of streamed-query-results) and then having streamed-query-results simply take a query arg and munge-fn. But this would require some pretty substantial changes to the extensions repo. Alternatively we could have two different `streamed-query-results` one which is used specifically for http stuff, and knows about http headers and executes the count query, and another which throws out the count query altogether for internal querying, like we need for sync. I wanted to take this intermediate step first to simplify things a little, if you think its just too complicated, I'm open to suggestions!